### PR TITLE
Fix issue where undefined is throw as connection error

### PR DESCRIPTION
**Pull request by swilliams-a3digital**
## Description

Brief description of the changes made in the PR. This helps in making better changelog
- 
- 
- 

# Reference to any github issues
-

### CODE REVIEW

**Comment by swilliams-a3digital** [comment]
Fix issue where undefined was thrown as exception

**Comment by swilliams-a3digital** [comment]
Addresses issue #26

**Comment by amarzavery** [comment]
@ramya-rao-a - If you are fine with this change, then please feel free to merge the PR.

File: lib/connection.ts [hunk 1]
```diff
@@ -242,7 +242,7 @@ export class Connection extends Entity {
 
         onClose = (context: RheaEventContext) => {
           removeListeners();
-          const err = context.error || context.connection.error;
+          const err = context.error || context.connection.error || Error('Failed to connect');
```
Comment (ramya-rao-a) [inline lib/connection.ts]: @amarzavery Won't we get the `onClose` event when the connection is closed explicitly?

Comment (ramya-rao-a) [reply lib/connection.ts]: Ah, never mind. All these listeners are removed as soon as the connection opens.

**Pull request merged**